src/tree/mod.rs

```diff
@@ -1134,6 +1248,10 @@ impl<T> Tree<T> {
     }
 }
 
+trait IteratorNew<'a, T, I> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> I;
+}
+
```

Comment:
If I'm not mistaken, this is the only way to make sure that the `new(...)` function is available the `tree` module, but not have it available any higher up than that.

Essentially I just wanted to be extra safe and not allow the library user to call `new(...)` on anything in the `iterators` module directly and this is the only pattern I could think of that would provide that behavior.